### PR TITLE
Fix build using mingw

### PR DESCRIPTION
**Pull request by nyalldawson**
Changes many  #if defined (Q_OS_WIN) to #ifdef _MSC_VER, where they relate specifically to msvc compiler. Submitted on behalf of Theuns Heydenrych.

### CODE REVIEW

File: src/core/qgis.h [hunk 1]
```diff
@@ -437,13 +437,12 @@ typedef unsigned long long qgssize;
 #endif
 
 #ifndef QGISEXTERN
-#ifdef Q_OS_WIN
+#ifdef _MSC_VER
```
Comment (nyalldawson) [inline src/core/qgis.h]: Theuns - I'm unsure about this change. Was this leftover from earlier testing? Didn't we establish that __declspec( dllexport ) was correct for the mingw build?

**Comment by NathanW2** [comment]
I didn't think we used mingw anymore?

**Comment by nyalldawson** [comment]
Not for the official builds... But if someone wants/needs to build with mingw, shouldn't we allow that?

**Comment by NathanW2** [comment]
I guess not as long as it doesn't lead to hack workarounds for something
unsupported IMO

File: src/core/qgis.h (hunk 1 above)
Comment (t-hey) [reply src/core/qgis.h]: Nyall - With the original code from the repo it used the `__GNUC__` define with the unix style deceleration and the plugins worked regarding loading data, I will have to make sure on Monday, thats why we changed the code like this.
You are right in the sense that `__declspec( dllexport )` is working correctly for all export functions.
So in this case `__declspec( dllexport )`  should work.

Comment (nyalldawson) [reply src/core/qgis.h]: For reference - off list discussion determined that this change was not necessary

**Pull request merged**